### PR TITLE
refactor init.sh in golang

### PR DESCRIPTION
**Pull request by bretagne-peiqi**
refactor init.sh in go implementation.
basic functions tested manually.
continuous optimization need to be made.

Signed-off-by: peiqi <uestc.shi@gmail.com>

### CODE REVIEW

**Comment by tgraf** [comment]
This looks great. As per comments in the code, there is some refactoring left to do to make all the file names be relative.

Other than that, it would be great to:
- Keep the NEWS entry
- use camelCase for all the variable names (this is where hound complained)
- Split the big main function into a couple smaller ones to improve readability of the code

File: daemon/daemon.go [hunk 1]
```diff
@@ -62,11 +59,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
-)
-
-const (
-	// ExecTimeout is the execution timeout to use in init.sh executions
-	ExecTimeout = time.Duration(30 * time.Second)
+	"github.com/cilium/cilium/pkg/bpfloader"
```
Comment (aanm) [inline daemon/daemon.go:62]: This should go between lines 32-54

File: daemon/daemon.go [hunk 2]
```diff
@@ -340,19 +333,9 @@ func (d *Daemon) compileBase() error {
...
-		log.Errorf("Command execution failed: Timeout for %s %s", prog, args)
-		return fmt.Errorf("Command execution failed: Timeout for %s %s", prog, args)
-	}
+	err := bpfloader.InitBPF(args...)
 	if err != nil {
-		log.Warningf("Command execution %s %s failed: %s", prog,
-			strings.Join(args, " "), err)
-		log.Warningf("Command output:\n%s", out)
-		return err
+		return fmt.Errorf("BPF base compile failed %s ", err)
```
Comment (aanm) [inline daemon/daemon.go:338]: necessary trailing space at the end of the error message.

**Review by aanm — COMMENTED** [review]
I'll continue the review after some @tgraf comments are addressed.

**Comment by bretagne-peiqi** [comment]
I close this one, because I made a mistake when repush the pr.

**Pull request closed without merging**